cpp/src/parquet/column_reader.cc

```diff
@@ -307,8 +307,15 @@ class SerializedPageReader : public PageReader {
   // Please refer to the encryption specification for more details:
   // https://github.com/apache/parquet-format/blob/encryption/Encryption.md#44-additional-authenticated-data
 
-  // The ordinal fields in the context below are used for AAD suffix calculation.
+  // The CryptoContext used by this PageReader.
   CryptoContext crypto_ctx_;
+  // This PageReader has its own copy of crypto_ctx_->meta_decryptor and
+  // crypto_ctx_->data_decryptor in order to be thread-safe. Do not mutate (update) the
+  // instances of crypto_ctx_.
```

Comment:
There are also uses of `UpdateDecryptor` in `RowGroupPageIndexReaderImpl`. I'm not sure whether that's used concurrently anywhere within Arrow code, but it's part of the public API so could be used concurrently by users and it would make sense to make a similar change there.

Reply:
Yes, they are used only once for a specific column.

Reply:
But the decryptor returned by `GetColumnMetaDecryptor` might be the `InternalFileDecryptor::footer_metadata_decryptor_` member in the case of uniform encryption, so although it's only used once, trying to use this method concurrently for different columns would risk race conditions where the AAD might be updated in another thread between being updated and used.